lib/node/nodes/buffer.js

```diff
@@ -122,17 +138,11 @@ Buffer.prototype.isolinesDissolvedQuery = function() {
 };
 
 Buffer.prototype.checkLimits = function(context, callback) {
-    var defaultLimit;
-    var limitName;
-    var limit;
+    var limits = this.dissolved ? LIMITS_DISSOLVED : LIMITS;
 
-    if (this.dissolved) {
-        limitName = 'maximumNumberOfRowsDissolved';
-        defaultLimit = 100000;
-    } else {
-        limitName = 'maximumNumberOfRows';
-        defaultLimit = 1000000;
-    }
-    limit = context.getLimit(TYPE, limitName, defaultLimit, 'too many source rows');
-    limits.limitInputRows(this, 'source', context, limit, callback);
+    Checks.check(this, context, [{
+        checker: Checks.limitInputRows,
```

Comment:
Isn't it a bit weird that the Checks.check function receives a reference to a function of the same module? I have the feeling this is still very _DSL_ish approach.

Why not something like:

```
this.limits.isWithinRowsLimit(this.source, limits.maximumNumberOfRows.default, callback)
```